packages/connect-core/src/connect-error.ts

```diff
@@ -74,9 +81,23 @@ export class ConnectError extends Error {
    */
   constructor(
     message: string,
-    code: Code = Code.Unknown,
+    code?: Code,
     metadata?: HeadersInit,
     outgoingDetails?: Message[]
+  );
+  constructor(
+    message: string,
+    code?: Code,
+    metadata?: HeadersInit,
+    outgoingDetails?: Message[],
+    cause?: unknown
+  );
+  constructor(
+    message: string,
+    code: Code = Code.Unknown,
+    metadata?: HeadersInit,
+    outgoingDetails?: Message[],
+    cause?: unknown
```

Comment:
Is it necessary to overload the signature? I would have expected we'd just do:

```diff
constructor(
    message: string,
    code: Code = Code.Unknown,
    metadata?: HeadersInit,
    outgoingDetails?: Message[],
+   cause?: unknown
```

Reply:
Ugh yeah.  Fixed this.  Think my earlier efforts at overloading confused the matter.